src/SonarScanner.MSBuild.Common/ConsoleLogger.cs

```diff
@@ -170,7 +154,7 @@ private void FlushOutput()
     /// Formats a message and writes or records it.
     /// </summary>
     private void Write(MessageType messageType, string message, params object[] args) =>
-        WriteFormatted(messageType, GetFormattedMessage(message, args));
+        WriteFormatted(messageType, FormatAndTimestampMessage(message, args));
```

Comment:
`FormatAndTimestampMessage` is called only here, which is what should happen, as the `Timestamp` should be added only once.

Another idea would be to remove `FormatAndTimestampMessage` ad in-line the logic here, so that no-one else can re-use it. 
Thoughts?